include/net/http.h

```diff
@@ -9,45 +9,388 @@
 
 #if defined(CONFIG_HTTP_CLIENT)
 
+#include <net/http_parser.h>
 #include <net/net_context.h>
 
+#define HTTP_CRLF "\r\n"
+
+/* Is there more data to come */
+enum http_final_call {
+	HTTP_DATA_MORE = 0,
+	HTTP_DATA_FINAL = 1,
+};
+
+#ifndef HTTP_PROTOCOL
+#define HTTP_PROTOCOL	   "HTTP/1.1"
+#endif
+
+/* Some generic configuration options, these can be overriden if needed. */
+#ifndef HTTP_STATUS_STR_SIZE
+#define HTTP_STATUS_STR_SIZE	32
+#endif
+
+/* Default network activity timeout in seconds */
+#define HTTP_NETWORK_TIMEOUT	K_SECONDS(20)
+
+/* It seems enough to hold 'Content-Length' and its value */
+#define HTTP_CONTENT_LEN_SIZE	48
+
+/* Default HTTP Header Field values for HTTP Requests if using the
+ * HTTP_HEADER_FIELDS define.
+ */
+#ifndef HTTP_ACCEPT
+#define HTTP_ACCEPT		"text/plain"
+#endif
+
+#ifndef HTTP_ACCEPT_ENC
+#define HTTP_ACCEPT_ENC		"identity"
+#endif
+
+#ifndef HTTP_ACCEPT_LANG
+#define HTTP_ACCEPT_LANG	"en-US"
+#endif
+
+#ifndef HTTP_CONNECTION
+#define HTTP_CONNECTION		"Close"
+#endif
+
+#ifndef HTTP_USER_AGENT
+#define HTTP_USER_AGENT	"Zephyr-HTTP-Client/1.8"
+#endif
+
+/* This can be used in http_client_send_get_req() when supplying
+ * extra_header_fields parameter.
+ */
+#ifndef HTTP_HEADER_FIELDS
+#define HTTP_HEADER_FIELDS				     \
+	"Accept: " HTTP_ACCEPT HTTP_CRLF		     \
+	"Accept-Encoding: " HTTP_ACCEPT_ENC HTTP_CRLF	     \
+	"Accept-Language: " HTTP_ACCEPT_LANG HTTP_CRLF	     \
+	"User-Agent: " HTTP_USER_AGENT HTTP_CRLF	     \
+	"Connection: " HTTP_CONNECTION HTTP_CRLF
+#endif
+
+struct http_client_ctx;
+
+/**
+ * @typedef http_receive_cb_t
+ * @brief Callback used when TCP data has been received from peer.
+ *
+ * @param ctx HTTP context.
+ * @param pkt Network packet.
+ */
+typedef void (*http_receive_cb_t)(struct http_client_ctx *ctx,
+				  struct net_pkt *pkt);
+
+/**
+ * @typedef http_request_cb_t
+ * @brief Callback used when reply has been received from peer.
```

Comment:
If it's "Callback used when reply has been received", why is it called http_**request**_cb_t?